### PR TITLE
Remove ArrivalOptions

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

We're discussing this at the moment. So opening this to test it.

### Changelog

```
<changelog>Remove `ArrivalOptions` in favor of triggering when the `RouteProgress.currentState` is equal to `RouteProgressState.COMPLETE`. You can use RouteProgress to implement custom arrival behavior based on time or distance.</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalObserver.kt [hunk 1]
```diff
@@ -12,11 +12,8 @@ import com.mapbox.navigation.core.MapboxNavigation
 interface ArrivalObserver {
 
     /**
-     * Called once the [ArrivalOptions] conditions have been met
-     * and the route progress has reached a waypoint on the route.
-     *
-     * [RouteProgress.currentState] has to be one of [RouteProgressState.LOCATION_TRACKING] or [RouteProgressState.ROUTE_COMPLETE]
-     * to trigger this event.
+     * Triggered when the [RouteProgress.currentState] is equal to [RouteProgressState.COMPLETE],
+     * once per route leg, and when there are more route legs to navigate.
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalObserver.kt:16]:
```suggestion
     * Triggered when the [RouteProgress.currentState] is equal to [RouteProgressState.COMPLETE],
     * once per route leg, and when there are more route legs to navigate. If we're on the last leg of the route [onFinalDestinationArrival] is called instead.
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalObserver.kt [hunk 2]
```diff
@@ -27,11 +24,8 @@ interface ArrivalObserver {
     fun onNextRouteLegStart(routeLegProgress: RouteLegProgress)
 
     /**
-     * Called once the [ArrivalOptions] conditions have been met
-     * and the route progress has reached the final destination on the route.
-     *
-     * [RouteProgress.currentState] has to be one of [RouteProgressState.LOCATION_TRACKING] or [RouteProgressState.ROUTE_COMPLETE]
-     * to trigger this event.
+     * Triggered when the [RouteProgress.currentState] is equal to [RouteProgressState.COMPLETE],
+     * once per route leg, and when there are no more route legs to navigate.
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalObserver.kt:28]:
```suggestion
     * Triggered when the [RouteProgress.currentState] is equal to [RouteProgressState.COMPLETE],
     * once when there are no more route legs to navigate.
```

**Review by LukasPaczos — COMMENTED** [review]
Could you add one more sentence to the changelog, like: "You can keep listening to `RouteProgress`' data (like distance/time remaining) to implement custom arrival behavior even when the state is already `COMPLETE`".

Might be worth squeezing this sentence into the `MapboxNavigation#registerArrivalObserver` and to `ArrivalObserver` interface as well.

**Comment by kmadsen** [comment]
🤔  failed the test. rebase and running again
```
snappingClosures=null}, voiceLanguage=en-US}>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.mapbox.navigation.instrumentation_tests.ui.routeline.AlternativeRouteSelectionTest$selectAlternateRoute$1$2.onRouteProgressChanged(AlternativeRouteSelectionTest.kt:127)
	at com.mapbox.navigation.core.trip.session.MapboxTripSession.updateRouteProgress(MapboxTripSession.kt:611)
	at com.mapbox.navigation.core.trip.session.MapboxTripSession.access$updateRouteProgress(MapboxTripSession.kt:58)
	at com.mapbox.navigation.core.trip.session.MapboxTripSession$updateDataFromNavigatorStatus$updateNavigatorStatusDataJob$1.invokeSuspend(MapboxTripSession.kt:581)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at androidx.test.espresso.base.Interrogator.loopAndInterrogate(Interrogator.java:148)
	at androidx.test.espresso.base.UiControllerImpl.loopUntil(UiControllerImpl.java:519)
	at androidx.test.espresso.base.UiControllerImpl.loopMainThreadUntilIdle(UiControllerImpl.java:441)
	at androidx.test.espresso.Espresso$1.run(Espresso.java:277)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Review by LukasPaczos — APPROVED** [review]

**Pull request merged**